modules/mysql/variables.tf

```diff
@@ -171,193 +213,249 @@ variable "read_replica_disk_type" {
 variable "failover_replica_disk_autoresize" {
   description = "Configuration to increase storage size."
+  type        = bool
   default     = true
 }
 
 variable "failover_replica_disk_size" {
   description = "The disk size for the failover replica instance."
+  type        = number
   default     = 10
 }
 
 variable "failover_replica_disk_type" {
   description = "The disk type for the failover replica instance."
+  type        = string
   default     = "PD_SSD"
 }
 
 variable "failover_replica_pricing_plan" {
   description = "The pricing plan for the failover replica instance."
+  type        = string
   default     = "PER_USE"
 }
 
 variable "failover_replica_replication_type" {
   description = "The replication type for the failover replica instance. Can be one of ASYNCHRONOUS or SYNCHRONOUS."
+  type        = string
   default     = "SYNCHRONOUS"
 }
 
 variable "failover_replica_database_flags" {
   description = "The database flags for the failover replica instance. See [more details](https://cloud.google.com/sql/docs/mysql/flags)"
+  type        = list(map(string))
   default     = []
 }
 
 variable "failover_replica_maintenance_window_day" {
   description = "The day of week (1-7) for the failover replica instance maintenance."
+  type        = number
   default     = 1
 }
 
 variable "failover_replica_maintenance_window_hour" {
   description = "The hour of day (0-23) maintenance window for the failover replica instance maintenance."
+  type        = number
   default     = 23
 }
 
 variable "failover_replica_maintenance_window_update_track" {
   description = "The update track of maintenance window for the failover replica instance maintenance. Can be either `canary` or `stable`."
+  type        = string
   default     = "canary"
 }
 
 variable "failover_replica_user_labels" {
+  type        = map(string)
   default     = {}
   description = "The key/value labels for the failover replica instance."
 }
 
 variable "failover_replica_ip_configuration" {
   description = "The ip configuration for the failover replica instances."
-
+  type = object({
+    authorized_networks = list(map(string))
+    ipv4_enabled        = bool
+    private_network     = string
+    require_ssl         = bool
+  })
   default = {
-    ipv4_enabled = "true"
+    authorized_networks = []
+    ipv4_enabled        = "true"
+    private_network     = null
+    require_ssl         = "false"
   }
 }
 
 variable "db_name" {
   description = "The name of the default database to create"
+  type        = string
   default     = "default"
 }
 
 variable "db_charset" {
   description = "The charset for the default database"
+  type        = string
   default     = ""
 }
 
 variable "db_collation" {
   description = "The collation for the default database. Example: 'utf8_general_ci'"
+  type        = string
   default     = ""
 }
 
 variable "additional_databases" {
   description = "A list of databases to be created in your cluster"
+  type        = list(map(string))
```

Comment:
Same because of:

		Schema: map[string]*schema.Schema{
			"instance": {
				Type:     schema.TypeString,
				Required: true,
				ForceNew: true,
			},
			"name": {
				Type:     schema.TypeString,
				Required: true,
				ForceNew: true,
			},
			"charset": {
				Type:     schema.TypeString,
				Computed: true,
				Optional: true,
			},
			"collation": {
				Type:     schema.TypeString,
				Computed: true,
				Optional: true,
			},
			"project": {
				Type:     schema.TypeString,
				Optional: true,
				Computed: true,
				ForceNew: true,
			},
			"self_link": {
				Type:     schema.TypeString,
				Computed: true,
			},
		},

Reply:
Fixed